core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/naming/generation/DefaultNameGenerator.java

```diff
@@ -8,11 +8,18 @@
 
   @Override
   public String generate(Nameable nameable) throws MisoNamingException {
+    DefaultMisoEntityPrefix selected = null;
     for (DefaultMisoEntityPrefix prefix : DefaultMisoEntityPrefix.values()) {
       if (prefix.getClass().isAssignableFrom(nameable.getClass())) {
-        return prefix.name() + nameable.getId();
+        if (selected != null) {
+          throw new MisoNamingException("Duplicate names for object of type: " + nameable.getClass().getSimpleName());
```

Comment:
Message is misleading. This isn't a duplicate name issue, but rather a "can't decide which name to use" situation